### DISCUSSION TITLE
Can this be used with JWT token based API authn and authz?

### DISCUSSION THREAD

**Discussion by toomanyjoes** [palkan/action_policy#148] [Q&A]
I've just briefly started looking at this library. I'm creating an authN authZ microservice so I was looking for an authoritative answer on whether this could be used in such a service using token based authentication to provide authorization via JWTs, or if it is primarily for session based authZ and authorization via JWTs is not recommended. Thanks!

**Comment by palkan** [comment]
Action Policy doesn't require any specific authentication method; all you need is to provide an [_authorization context_](https://actionpolicy.evilmartians.io/#/authorization_context) (e.g., `current_user` or whatever), and it's up to you how to retrieve it from the request.
